### PR TITLE
IGNITE-14352 Add snapshot check command for custom cache groups

### PR DESCRIPTION
**Pull request by Mmuzaf**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyTaskResult.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import java.io.ObjectOutput;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.apache.ignite.cluster.ClusterNode;
+import org.apache.ignite.internal.dto.IgniteDataTransferObject;
+import org.apache.ignite.internal.processors.cache.verify.IdleVerifyResultV2;
+import org.apache.ignite.internal.util.typedef.internal.U;
+
+public class SnapshotPartitionsVerifyTaskResult extends IgniteDataTransferObject {
```
Comment (xtern) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyTaskResult.java:31]: javadoc is missing

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotRestoreProcess.java [hunk 2]
```diff
@@ -184,90 +184,84 @@ protected void cleanup() throws IgniteCheckedException {
...
+                return;
+            }
 
-                Set<UUID> dataNodes = new HashSet<>();
-                Set<String> snpBltNodes = null;
-                Map<ClusterNode, List<SnapshotMetadata>> metas = f.result();
-                Map<Integer, String> reqGrpIds = cacheGrpNames == null ? Collections.emptyMap() :
-                    cacheGrpNames.stream().collect(Collectors.toMap(CU::cacheId, v -> v));
+            if (!F.isEmpty(f.result().exceptions())) {
+                finishProcess(fut0.rqId, F.firstEntry(f.result().exceptions()).getValue());
```
[the remark this answers is not in the document]
Comment (Mmuzaf) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotRestoreProcess.java]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyTask.java [hunk 3]
```diff
@@ -165,13 +175,19 @@
         /** Consistent snapshot metadata file name. */
         private final String consId;
 
+        /** Set of cache groups to be checked in the snapshot or {@code empty} to check everything. */
+        private final Set<String> rqGrps = new HashSet<>();
+
         /**
          * @param snpName Snapshot name to validate.
          * @param consId Consistent snapshot metadata file name.
          */
```
[the remark this answers is not in the document]
Comment (Mmuzaf) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyTask.java:186]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyTaskResult.java (hunk 1 above)
Comment (Mmuzaf) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyTaskResult.java:35]: Fixed.

**Review by xtern — APPROVED** [review]

**Pull request merged**